src/diffusers/models/unet_2d_condition.py

```diff
@@ -310,8 +310,12 @@ def forward(
         timesteps = timestep
         if not torch.is_tensor(timesteps):
             # TODO: this requires sync between CPU and GPU. So try to pass timesteps as tensors if you can
-            timesteps = torch.tensor([timesteps], dtype=torch.long, device=sample.device)
-        elif torch.is_tensor(timesteps) and len(timesteps.shape) == 0:
```

Comment:
I think `torch.is_tensor(timesteps)` was redundant here.

Reply:
Hmm actually there might still be use cases where `timesteps` are just int values - could we maybe instead deprecate it?

Reply:
I just changed the `elif` condition to check `len(timesteps.shape)`, because `not torch.is_tensor` is the `if` branch. The comment inside the diff makes it confusing.